### ISSUE TITLE
Scope `BertSpanLabeler` and supporting infra

### ISSUE THREAD

**Issue by jbischof** [keras-team/keras-nlp#625]
`BertClassifier` supports one output for the entire input sequence, but span labeling is an important research benchmark. This will require some scoping and design:

1. Do we need a separate `BertSpanLabelingPreprocessor` or just a more complicated label to pass through?
2. Do we want to offer a script that can turn a standard dataset like [squadv2](https://www.tensorflow.org/datasets/catalog/squad) into something our preprocessor can use? The "answer text" must be turned into indicies in the tokenized input.
3. Create a `BertSpanLabeler` task model with the correct task-specific layers and preprocessing.

**Comment by jbischof** [comment]
References: 
* `tensorflow/models` has `BertSpanLabeler`: https://www.tensorflow.org/tfmodels/nlp#span_labeling_model
* HF has `BertForQuestionAnswering`: https://huggingface.co/docs/transformers/model_doc/bert#transformers.BertForQuestionAnswering
